### DISCUSSION TITLE
Missing blocks after compaction

### DISCUSSION THREAD

**Discussion by madhaviuddaraju** [cortexproject/cortex#5418] [General]
**Describe the bug**
 I'm experiencing issues with the compactor in 1.15.1 version. it doesn't seem to be working correctly. I've noticed that some blocks are missing after the compaction process. **To Reproduce**

1. Cortex 1.1.5.1
2. Compactor

**Expected behavior**
After compaction process also , i should able to see the old data when queried for duration , graph shows gaps.

**Environment:**
 - Infrastructure:Kubernetes
 - Deployment tool: helm
**Additional Context**

**Comment by alanprot** [comment]
How is the `deletion_delay ` and the `cleanup_interval ` are configured?

**Comment by madhaviuddaraju** [comment]
Thanks @alanprot 
Below are the configured values
cleanup_interval: 15m0s
deletion_delay: 12h0m0s

**Comment by alanprot** [comment]
Do you have bucket index enabled? if not, can you enable it?

**Comment by madhaviuddaraju** [comment]
yes , we have enabled bucket index.
@alanprot I am facing an issue that after compaction to 24 hours block , we are seeing missing blocks , any configs i need to cross check to fix this?
I have deleted the configs related "out_of_order_time_window: 4h", is this config causing missing blocks after compaction?

**Comment by alanprot** [comment]
oh.. you are using the OOTO feature?

If so, i think this is the problem: https://github.com/cortexproject/cortex/issues/5402
